internal/build/tar_test.go

```diff
@@ -17,18 +17,22 @@ import (
 
 func TestArchiveDf(t *testing.T) {
 	f := newFixture(t)
-	ab := NewArchiveBuilder(model.EmptyMatcher)
-	defer ab.close()
-	defer f.tearDown()
 
 	dfText := "FROM alpine"
-	df := dockerfile.Dockerfile(dfText)
-	err := ab.archiveDf(f.ctx, df)
-	if err != nil {
-		t.Fatal(err)
-	}
+	pr, pw := io.Pipe()
+	go func() {
+		ab := NewArchiveBuilder(pw, model.EmptyMatcher)
+		defer ab.Close()
+		defer f.tearDown()
+
+		df := dockerfile.Dockerfile(dfText)
+		err := ab.archiveDf(f.ctx, df)
+		if err != nil {
+			t.Fatal(err)
```

Comment:
you can't use t.Fatal in a goroutine. it won't work correctly and weird things will happen

https://golang.org/pkg/testing/#T